astroid/inference.py

```diff
@@ -1025,13 +1030,15 @@ def infer_assign(self, context=None):
     return bases._infer_stmts(stmts, context)
 
 
-nodes.AssignName._infer = infer_assign
-nodes.AssignAttr._infer = infer_assign
+nodes.AssignName._infer = infer_assign  # type: ignore[assignment]
+nodes.AssignAttr._infer = infer_assign  # type: ignore[assignment]
```

Comment:
Forgot to say it on previous merge request but I guess this is done in order to fight against circular import. Did you try to define this function directly in ``AssignAttr`` ? For some function I think it's done by force of habits more than necessity. This would simplify the code a lot and this kind of refactor worked in the past (although it's a bet and uncover big issue sometime).

Reply:
No I haven't tried this. However, it would be very difficult to reuse functions like we do here.

The best refactor would be to create Mixins/Base classes with the current inference functions that can then be inherited, but that's something for 2043 😄

Reply:
Let's not do that, at least not for a while. True, it might be possible to move some methods but that can just lead to more issues down the road. What if for example we  want to split up `node_classes.py` and suddenly we added a new circular import just because of it.

Furthermore, although that might just be might opinion, I find it quite useful to have most of the inference methods in one place. Especially considering the nodes file is huge.

Reply:
Right, I did not think of this reusing problem. It was used once in the  example I saw.